compiler/README.md

```diff
@@ -36,3 +42,136 @@ To generate Java interfaces with protobuf nano:
 $ protoc --plugin=protoc-gen-java_rpc=build/binaries/java_pluginExecutable/java_plugin \
   --java_rpc_out=nano=true:"$OUTPUT_FILE" --proto_path="$DIR_OF_PROTO_FILE" "$PROTO_FILE"
 ```
+
+## Installing the codegen to Maven local repository
+This will compile a codegen and put it under your ``~/.m2/repository``. This
+will make it available to any build tool that pulls codegens from Maven
+repostiories.
+```
+$ ../gradlew install
+```
+
+## Pushing the codegen to Maven Central
+This will compile both the 32-bit and 64-bit versions of the codegen and upload
+them to Maven Central.
+
+You need to have both the 32-bit and 64-bit versions of Protobuf installed.
+It's recommended that you install Protobuf to custom locations (e.g.,
+``~/protobuf-3.0.0-32`` and ``~/protobuf-3.0.0-64``) rather than the system
+default location, so that different version of Protobuf won't mess up. It
+should also be compiled statically linked.
+
+Generally speaking, to compile and install both 32-bit and 64-bit artifacts locally:
+```
+$ TARGET_ARCHS="<list-of-archs>" ../gradlew install --info
+```
+
+To compile and upload the artifacts to Maven Central:
+```
+$ TARGET_ARCHS="<list-of-archs>" ../gradlew uploadArchives
+```
+
+It's recommended that you run ``install --info`` prior to ``uploadArchives``
+and examine the produced artifacts before running ``uploadArchives``.
+
+Additional environments need to be set so that the compiler and linker can find
+the protobuf headers and libraries. Follow the platform-specific instructions
+below.
+
+### Linux
+You must have multilib GCC installed on your system.
+
+Compile and install 32-bit protobuf:
+```
+protobuf$ CXXFLAGS="-m32" ./configure --disable-shared --prefix=$HOME/protobuf-32
+protobuf$ make clean && make -j && make install
```

Comment:
We probably don't want to have "-j" there. That will exhaust the memory of some machines if I'm reading make's documentation correctly.